[v0.10] cherry picks

metalefty opened the pull request:
Backport #2954, #2955, #2956, #2962, #2967, #2970, #2972.

Other additional pull requests might be cherry-picked later. I'll keep this open for a while.

metalefty committed:
GFX: Relegate some logs to LOG_LEVEL_DEBUG

(cherry picked from commit 7e305f9e2421f48b61e7f56c0c474f4b275ccc2f)

matt335672 committed:
Migrate github actions to Node 20

Github actions are transitioning to Node 20 ans Node 16 is EOL.

https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20
(cherry picked from commit 4ba026be3842194faa55b3af811780a6b4051eed)

metalefty committed:
Bump copyright year and make easier to bump

(cherry picked from commit ae249c675542092b4487ce040cb6fc06d43f1e87)

metalefty commented:
It might be better to use `git cherry-pick -x` for cherry-picking. 

```
       -x
           When recording the commit, append a line that says "(cherry picked
           from commit ...)" to the original commit message in order to
           indicate which commit this change was cherry-picked from. This is
           done only for cherry picks without conflicts. Do not use this
           option if you are cherry-picking from your private branch because
           the information is useless to the recipient. If on the other hand
           you are cherry-picking between two publicly visible branches (e.g.
           backporting a fix to a maintenance branch for an older release from
           a development branch), adding this information can be useful.
```

matt335672 commented:
Can you add #2955 ? It will be needed for any CI when Node 16 gets turned off in a couple of months?

I was just about to add individual PRs for these but I can't see any advantage to that.

I didn't know about `cheey-pick -x`. Thanks for commenting on that.

metalefty commented:
Added!

rowlap committed:
Remove tcutils channel from xrdp.ini

According to #1943 tcutils was removed, so update the channel section to match.

(cherry picked from commit 2a0c2a612f61ed98f86901b431edc44941e4b197)

metalefty commented:
I learned `cherry-pick -x` from FreeBSD comitter's guide. It is used in MFC (Merge from current), very similar workflow to our backporting.

https://docs.freebsd.org/en/articles/committers-guide/#mfc-with-git

metalefty commented on configure.ac:
```diff
@@ -1,7 +1,9 @@
 AC_PREREQ([2.69])
-AC_INIT([xrdp], [0.9.90], [xrdp-devel@googlegroups.com])
+=======
+AC_INIT([xrdp], [0.10.80], [xrdp-devel@googlegroups.com])
```
Oops, this should be 0.9.90.

matt335672 committed:
GFX: Fix disconnect on resize of busy windows

When a resize is underway on a busy X server, it is possible for a
queued EGFX cmd (order #62) to be processed after the decoder has been
deleted. This causes a client disconnect with no useful error message.

(cherry picked from commit 3430b8898c8f95b31af72a58a8eeafeae65640d5)

derekschrock committed:
Fall back to IPv4 if IPv6 capable but don't have an IPv6 address set

When xrdp is built with IPv6 support it will only fall back to IPv4 if
IPv6 is not supported (EAFNOSUPPORT).  However, if the system is IPv6
capable but doesn't have an IPv6 address set (at least inside a FreeBSD
jail) EPROTONOSUPPORT is returned from socket().

(cherry picked from commit 5afbca4954cf585805c80e72620183914a8719f6)

matt335672 committed:
Increase GFX output buffer size

Size the GFX output buffer pessimistically based on the largest monitor
in the configuration.

(cherry picked from commit 90e4aca26a7f256b93a31168da3a2682d0ca22df)

metalefty committed:
Suppress some logs when no errors

This outputs per-frame logs when using VNC backend and on login screen.

(cherry picked from commit 0b4150ff488ab188fd5610a1ae2382fca5153d3b)

metalefty commented:
Merging this.
